### PR TITLE
feat: moved sharable contacts code to commons

### PR DESCRIPTION
**Pull request by ismailnurudeen**

### CODE REVIEW

File: app/src/main/kotlin/com/simplemobiletools/contacts/pro/databases/ContactsDatabase.kt [hunk 1]
```diff
@@ -1,90 +1,90 @@
...
-        private val MIGRATION_2_3 = object : Migration(2, 3) {
-            override fun migrate(database: SupportSQLiteDatabase) {
-                database.apply {
-                    execSQL("ALTER TABLE contacts ADD COLUMN ringtone TEXT DEFAULT ''")
-                }
-            }
-        }
-    }
-}
+//package com.simplemobiletools.contacts.pro.databases
```
Comment (tibbi) [inline app/src/main/kotlin/com/simplemobiletools/contacts/pro/databases/ContactsDatabase.kt:1]: do not add such useles files...

**Comment by tibbi** [comment]
why are so many files fully commented out?

**Comment by ismailnurudeen** [comment]
> why are so many files fully commented out?

They will be removed, I have converted this PR to a draft... I'm still testing some stuff.

File: app/src/main/kotlin/com/simplemobiletools/contacts/pro/activities/EditContactActivity.kt [hunk 2]
```diff
@@ -727,7 +744,7 @@ class EditContactActivity : ContactActivity() {
     }
 
     private fun setupNewContact() {
-        originalContactSource = if (hasContactPermissions()) config.lastUsedContactSource else SMT_PRIVATE
+        originalContactSource = if (hasContactPermissions()) contactsConfig.lastUsedContactSource else SMT_PRIVATE
```
Comment (tibbi) [inline app/src/main/kotlin/com/simplemobiletools/contacts/pro/activities/EditContactActivity.kt:747]: as discussed, use the old config instead of the new contactsConfig

File: app/src/main/kotlin/com/simplemobiletools/contacts/pro/helpers/VcfExporter.kt [hunk 3]
```diff
@@ -176,29 +171,29 @@ class VcfExporter {
...
     private fun getPhoneNumberTypeLabel(type: Int, label: String) = when (type) {
-        Phone.TYPE_MOBILE -> CELL
-        Phone.TYPE_HOME -> HOME
-        Phone.TYPE_WORK -> WORK
-        Phone.TYPE_MAIN -> PREF
-        Phone.TYPE_FAX_WORK -> WORK_FAX
-        Phone.TYPE_FAX_HOME -> HOME_FAX
-        Phone.TYPE_PAGER -> PAGER
-        Phone.TYPE_OTHER -> OTHER
+        ContactsContract.CommonDataKinds.Phone.TYPE_MOBILE -> CELL
```
Comment (tibbi) [inline app/src/main/kotlin/com/simplemobiletools/contacts/pro/helpers/VcfExporter.kt:174]: these are unnecessary changes, just readd the old imports

File: app/src/main/kotlin/com/simplemobiletools/contacts/pro/activities/EditContactActivity.kt [hunk 4]
```diff
@@ -727,7 +744,7 @@ class EditContactActivity : ContactActivity() {
     }
 
     private fun setupNewContact() {
-        originalContactSource = if (hasContactPermissions()) config.lastUsedContactSource else SMT_PRIVATE
+        originalContactSource = if (hasContactPermissions()) this.baseConfig.lastUsedContactSource else SMT_PRIVATE
```
Comment (tibbi) [inline app/src/main/kotlin/com/simplemobiletools/contacts/pro/activities/EditContactActivity.kt:747]: no need for "this", just use it as it was

File: app/src/main/kotlin/com/simplemobiletools/contacts/pro/activities/EditContactActivity.kt [hunk 5]
```diff
@@ -378,7 +388,7 @@ class EditContactActivity : ContactActivity() {
     }
 
     private fun setupFieldVisibility() {
-        val showFields = config.showContactFields
+        val showFields = baseConfig.showContactFields
```
Comment (tibbi) [inline app/src/main/kotlin/com/simplemobiletools/contacts/pro/activities/EditContactActivity.kt:391]: theres still a lot of these useless changes in the code... Just look through the whole code this PR changes on github, you can see irrelevant changes easily

Comment (ismailnurudeen) [reply app/src/main/kotlin/com/simplemobiletools/contacts/pro/activities/EditContactActivity.kt:391]: the `config` to `baseConfig` change here is actually valid as the entire config in contacts is now being shared via commons using the `baseConfig`.

**Comment by tibbi** [comment]
alright, guess it should be good now, thanks :)
